lib/dataTrigger/dpnsTriggers/createDomainDataTrigger.js

```diff
@@ -111,6 +111,17 @@ async function createDomainDataTrigger(documentTransition, context, topLevelIden
       return result;
     }
 
+    if (subdomainRules.allowSubdomains === true) {
+      result.addError(
+        new DataTriggerConditionError(
+          documentTransition,
+          context.getDataContract(),
+          context.getOwnerId(),
+          'Allowing subdomains registration is forbidden for non top level domains',
+        ),
+      );
+    }
+
     if (subdomainRules.allowSubdomains === false
```

Comment:
We can return the result after adding that error and remove `subdomainRules.allowSubdomains === false` condition.